internal/controller/datadogagent/merger/rbac.go

```diff
@@ -22,7 +22,6 @@ type RBACManager interface {
 	AddServiceAccount(namespace string, name string) error
 	AddServiceAccountByComponent(namespace, name, component string) error
 	AddServiceAccountAnnotations(namespace string, name string, annotations map[string]string) error
-	AddServiceAccountAnnotationsByComponent(namespace string, name string, annotations map[string]string, component string) error
```

Comment:
The `ByComponent` functions are used to delete rbac in `override.go`. I don't believe this is needed since we've got `AddServiceAccountByComponent`